mesonbuild/wrap/wrap.py

```diff
@@ -515,6 +517,25 @@ def get_svn(self) -> None:
         subprocess.check_call([svn, 'checkout', '-r', revno, self.wrap.get('url'),
                                self.directory], cwd=self.subdir_root)
 
+    def get_netrc_credentials(self, netloc: str) -> T.Optional[T.Tuple[str, str]]:
+        try:
+            rc = netrc()
+        except NetrcParseError as e:
+            mlog.warning(f'failed to parse netrc file: {e}.', fatal=False)
+            return None
+        except Exception as e:
```

Comment:
I think you should catch only `FileNotFoundError`, and I don't think it's worth loggin the error, just ignore and return None, IMHO.

Reply:
We can also just hardcode the filename we know netrc is going to use (that is, `os.path.expanduser('~/.netrc')`) and return early if it isn't available. It's going to be unavailable more often than not, I suspect, so exceptions may not be the most elegant approach.

Reply:
Yes but if it's 600 then the actual user in question can still read it...

If there's a permission error then arguably something went wrong and we maybe should error, or at least warn about it.

If the file simply doesn't exist then there's nothing wrong ***and*** we should not warn about it.